### PR TITLE
Feature/fixs

### PR DESCRIPTION
**Pull request by nathanmalishev**
Addressing these issues
- restore returns array of keys restored
- update readme
- make the backup file json
- standard date formatter for timestamp
- tests (work in progress!)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,36 +14,39 @@ npm install consul-backup-restore
...
 )
 ```
-Callbacks only argument is error
+Callbacks has an error argument and a result argument. The result argument will either be the name of the file, or the s3 data dump.
 
 Options
-* prefix : Consul prefix used to back up keys from consul. If left blank all key value pairs will be backed up
-* s3_bucket_name: If you wish to use AWS's s3 bucket, specify the buckets name
+* prefix : Consul prefix used to back up keys from consul. If left blank all key value pairs will be backed up.
+* s3_bucket_name: If you wish to use AWS's s3 bucket, specify the buckets name.
```
Comment (markgandy) [inline README.md]: I think this one is s3BucketName

Comment (nathanmalishev) [inline README.md]: Correct! changed it

File: src/consulUtil.js [hunk 2]
```diff
@@ -25,21 +25,24 @@ exports.getKeyValues = function (consulInstance, prefix, callback) {
...
-    const key = JSON.parse(kv).Key
-    const backupValue = JSON.parse(kv).Value
+  var keyValues = JSON.parse(rawData.toString('utf-8'))
 
-    getConsulKey(key)
-           .then((consulValue) => overrideKey(key, consulValue, override))
-           .then(() => setConsulKeyValue(key, backupValue))
-           .then((restoredKey) => callback(null, `Key ${restoredKey} was restored`))
-           .catch((e) => callback(e))
+  async.map(keyValues, setConsulAndCheckOverride.bind(override), (err, kv) => {
```
Comment (markgandy) [inline src/consulUtil.js]: This is nice! Only comment is that kv var here is an array of keys yeah? Maybe rename to keys

**Comment by markgandy** [comment]
Can we also update the version to v1.2.0 ?

File: src/consulUtil.js (hunk 2 above)
Comment (nathanmalishev) [inline src/consulUtil.js]: Yeah array of keys, rename incoming

**Pull request merged**